### PR TITLE
shard doc improvements

### PR DESCRIPTION
**Pull request by SeanNaren**
## What does this PR do?

Adds a bit more info to the shard docs! I've also updated the link so that if people install it actually works. We'll be updating this again in the pipe docs, but in the interim let's give the right link :)

Fixes # (issue)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/multi_gpu.rst [hunk 1]
```diff
@@ -611,38 +611,71 @@ This is useful when dealing with large Transformer based models, or in environme
...
-This is primarily because with larger batch sizes, storing activations for the backwards pass becomes the bottleneck in training. Sharding optimizer state as a result becomes less impactful.
+**Increase Your Batch Size**
 
-To use Optimizer Sharded Training, you need to first install Fairscale using the command below or install all extras using ``pip install pytorch-lightning["extra"]``.
+Use Sharded Training to scale your batch size further using the same compute. This will reduce your overall epoch time.
+Below we use the `NeMo Transformer Lightning Language Modeling example <https://github.com/NVIDIA/NeMo/tree/main/examples/nlp/language_modeling>`_ to benchmark the maximum batch size that can be fit on 8 A100 GPUs for DDP vs Sharded Training.
+Note that the benefits can still be obtained using 2 or more GPUs, and for even larger batch sizes you can scale to multiple nodes.
+
++----------------------+-----------------------+----------------+
+| Distributed Training | Model Size (Millions) | Max Batch Size |
```
Comment (tchaton) [inline docs/source/multi_gpu.rst:637]: Can you add % gain ?

File: docs/source/multi_gpu.rst [hunk 2]
```diff
@@ -611,38 +611,71 @@ This is useful when dealing with large Transformer based models, or in environme
 
 Lightning currently offers the following methods to leverage model parallelism:
 
-- Optimizer Sharded Training (partitioning your gradients and optimizer state across multiple GPUs, for reduced memory overhead)
+- Sharded Training (partitioning your gradients and optimizer state across multiple GPUs, for reduced memory overhead with **no performance loss**)
 
-Optimizer Sharded Training
-^^^^^^^^^^^^^^^^^^^^^^^^^^
-Lightning integration of optimizer sharded training provided by `Fairscale <https://github.com/facebookresearch/fairscale>`_.
+Sharded Training
```
Comment (tchaton) [inline docs/source/multi_gpu.rst:616]: Would you mind adding some images and maybe even put a link to your blogpost when released.

**Review by tchaton — APPROVED** [review]

File: docs/source/multi_gpu.rst (hunk 2 above)
Comment (SeanNaren) [reply docs/source/multi_gpu.rst:616]: Absolutely, will do in a followup PR after release :)

File: docs/source/multi_gpu.rst (hunk 1 above)
Comment (SeanNaren) [reply docs/source/multi_gpu.rst:637]: Will do!

File: docs/source/multi_gpu.rst [hunk 3]
```diff
@@ -611,38 +611,71 @@ This is useful when dealing with large Transformer based models, or in environme
...
+Use Sharded Training to scale your batch size further using the same compute. This will reduce your overall epoch time.
+Below we use the `NeMo Transformer Lightning Language Modeling example <https://github.com/NVIDIA/NeMo/tree/main/examples/nlp/language_modeling>`_ to benchmark the maximum batch size that can be fit on 8 A100 GPUs for DDP vs Sharded Training.
+Note that the benefits can still be obtained using 2 or more GPUs, and for even larger batch sizes you can scale to multiple nodes.
+
++----------------------+-----------------------+----------------+---------------------+
+| Distributed Training | Model Size (Millions) | Max Batch Size | Percentage Gain (%) |
++======================+=======================+================+=====================+
+| DDP                  | 930                   | 32             | -                   |
++----------------------+-----------------------+----------------+---------------------+
+| Sharded              | 930                   | **52**         | **48%**             |
```
Comment (Borda) [inline docs/source/multi_gpu.rst:641]: Sharded is also DDP, right? so rather "native DDP" and "Sharded DDP"

File: docs/source/multi_gpu.rst [hunk 4]
```diff
@@ -611,38 +611,71 @@ This is useful when dealing with large Transformer based models, or in environme
...
++----------------------+-----------------------+----------------+---------------------+
+
+**Increase Your Model Size**
+
+Use Sharded Training to scale your model size further using the same compute.
+We use the `NeMo Transformer Lightning Language Modeling example <https://github.com/NVIDIA/NeMo/tree/main/examples/nlp/language_modeling>`_ to benchmark the maximum model size that can fit on 8 A100 GPUs for DDP vs Sharded Training.
+Note that the benefits can still be obtained using 2 or more GPUs, and for even larger models you can scale to multiple nodes.
```
Comment (Borda) [inline docs/source/multi_gpu.rst:648]: is it exactly the same as above?

File: docs/source/multi_gpu.rst [hunk 5]
```diff
@@ -611,38 +611,71 @@ This is useful when dealing with large Transformer based models, or in environme
...
+| DDP                  | 32         | 930                       | -                   |
++----------------------+------------+---------------------------+---------------------+
+| Sharded              | 32         | **1404**                  | **41%**             |
++----------------------+------------+---------------------------+---------------------+
+| DDP                  | 8          | 1572                      | -                   |
++----------------------+------------+---------------------------+---------------------+
+| Sharded              | 8          | **2872**                  | **59%**             |
++----------------------+------------+---------------------------+---------------------+
+
+It is highly recommended to use Sharded Training in multi-GPU environments where memory is limited, or where training larger models are beneficial (500+ million parameter models).
```
Comment (Borda) [inline docs/source/multi_gpu.rst:662]:
```suggestion
It is highly recommended to use Sharded Training in multi-GPU environments where memory is limited, or where training larger models are beneficial (500M+ parameter models).
```

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

**Pull request merged**